src/ripple/basics/base_uint.h

```diff
@@ -419,7 +419,7 @@ class base_uint
 
     base_uint<Bits, Tag>& operator=(Zero)
     {
-        memset (&pn[0], 0, sizeof (pn));
+        memset (pn.data (), 0, sizeof (pn));
```

Comment:
This can now be:
```
pn.fill(0);
```

Reply:
@nbougalis Sorry for the long latency response w/ CppNow going on.  As far I can tell,

* We can remove `compare` in `base_uint.h` and delegate like you suggested just by making the comparison ops friends.
* For fixing the call sites that were relying on this `compare` function being available to define their own ordering (such as `Issue.cpp`), we can then just use the comparison operators directly.

This weekend, I should have some free time to clean up this PR and I'll post a proposed patch then.

Reply:
After digging into this for a little bit, I'd prefer to punt on this for now.  Implementing the comparison operators just as delegating to the `std::array` comparison ops didn't "just work".  Plus, it looks like we pass this `compare` function around as a function object in a different place (`TxMeta.cpp`).

So, my preference would be that I'll create a GitHub issue to capture this idea but do not wish to couple it with this PR, even though it may naturally seem like a good candidate.